File: app/src/main/java/com/sachtech/datingapp/presenter/UserPresenter.java
/*
package com.sachtech.datingapp.presenter;

import com.sachtech.datingapp.data.User;
import com.sachtech.datingapp.model.UserModel;
import com.sachtech.datingapp.view.UserView;
import kotlin.Lazy;
import kotlin.LazyKt;
import kotlin.Metadata;
import kotlin.jvm.functions.Function0;
import kotlin.jvm.functions.Function1;
import kotlin.jvm.internal.Intrinsics;
import kotlin.jvm.internal.PropertyReference1Impl;
import kotlin.jvm.internal.Reflection;
import kotlin.reflect.KProperty;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;


public final class UserPresenter {
   @NotNull
   private final UserView userView;

   @NotNull
   public final UserModel getUserModel() {
      return new UserModel();
   }

   public final void getAllUsers() {
      this.getUserModel().getAllUsers(this.userView);
   }

   public final void addToLikeList(@NotNull User user) {
      this.getUserModel().addUserToLikeList(user, this.userView);
   }

   public final void addToVisitList(@NotNull User user) {
      this.getUserModel().addToVisitList(user, this.userView);
   }

   public final void notifyLikedUser(@NotNull User user) {
      this.getUserModel().notifyLikedUser(user);
   }

   public final void addToChatFriendList(@NotNull User user, @NotNull OnFriendAdd onFreindAdd) {
      this.getUserModel().addUserToFriendList(user, onFreindAdd);
   }

   void OnFriendAdd(){

   }
   public final void addToUnlikeList(@NotNull User user) {
      this.getUserModel().addUserToUnlikeList(user, this.userView);
   }

   public final void removeFromUnlikeList(@NotNull User user) {
      Intrinsics.checkParameterIsNotNull(user, "user");
      this.getUserModel().removeUserFromUnliekeList(user);
   }

   public final void removeFromLikeList(@Nullable String docId) {
      this.getUserModel().removeUserFromLikeList(docId);
   }

   @NotNull
   public final UserView getUserView() {
      return this.userView;
   }

   public UserPresenter(@NotNull UserView userView) {
      super();
      this.userView = userView;
   }
}
*/

File: app/src/main/java/com/sachtech/datingapp/presenter/UserPresenter.kt
package com.sachtech.datingapp.presenter

import com.sachtech.datingapp.data.User
import com.sachtech.datingapp.model.UserModel
import com.sachtech.datingapp.ui.home.adapter.listener.OnFriendAdded
import com.sachtech.datingapp.view.UserView

class UserPresenter(val userView: UserView) {

    val userModel by lazy { UserModel() }
    fun getAllUsers() {
        userModel.getAllUsers(userView)
    }


    fun addToLikeList(user: User) {
        userModel.addUserToLikeList(user, userView)
    }

    fun addToVisitList(user: User) {
        userModel.addToVisitList(user, userView)
    }


/* fun addToBlockList(uid: User) {

         userModel.addUserToBlockList(uid,userView)
    }*/


    fun notifyLikedUser(user: User) {
        userModel.notifyLikedUser(user)
    }

    fun addToChatFriendList(user: User, onFreindAdd: OnFriendAdded) {
        userModel.addUserToFriendList(user, onFreindAdd)
    }

    fun addToUnlikeList(user: User) {
        userModel.addUserToUnlikeList(user, userView)
    }

    fun removeFromUnlikeList(user: User) {
        userModel.removeUserFromUnliekeList(user)
    }

    fun removeFromLikeList(docId: String?) {
        userModel.removeUserFromLikeList(docId)
    }


}
